### ISSUE TITLE
irgen: Unnecessary instructions generated for boolean expression

### ISSUE THREAD

**Issue by mewmew** [mewmew/uc#66]
```shell
uclang testdata/quiet/lexer/l04.c
```

* Remove zext from conditional binary expressions, so that m.cond may
correctly identify the boolean value.

```diff
-%8 = icmp eq i32 1, 0
-%9 = zext i1 %8 to i32
-%10 = icmp ne i32 %9, 0
+%8 = icmp eq i32 1, 0
```

**Comment by mewmew** [comment]
Marked as a future ambition for now.

**Comment by mewmew** [comment]
Fixed as of revision a62d7fc7077bc1d89c6a99685263d2820c64f619.
